src/Processor/Braintree/Query/SubscriptionQuery.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace TeamGantt\Dues\Processor\Braintree\Query;
+
+use Braintree\Gateway;
+use Braintree\SubscriptionSearch;
+use TeamGantt\Dues\Exception\InvalidSubscriptionSearchParamException;
+use TeamGantt\Dues\Model\Subscription;
+use TeamGantt\Dues\Processor\Braintree\Mapper\SubscriptionMapper;
+
+class SubscriptionQuery
+{
+    private Gateway $gateway;
+
+    private SubscriptionMapper $mapper;
+
+    /**
+     * @var mixed[] Items appended must be supported search methods found in `\Braintree\SubscriptionSearch`
+     */
+    private array $searchParams = [];
+
+    public function __construct(Gateway $gateway, SubscriptionMapper $mapper)
+    {
+        $this->gateway = $gateway;
+        $this->mapper = $mapper;
+    }
+
+    public function daysPastDue(string $comparison, int $days): self
```

Comment:
I'm such a sucker for prefixing these sorts of things with `where`. Are you not keen on `whereDaysPastDue`?